mittab/apps/tab/pairing_views.py

```diff
@@ -105,6 +107,100 @@ def assign_judges_to_pairing(request):
+    round_obj = Round.objects.get(id=int(round_id))
+    round_number = round_obj.round_number
+
+    current_room_obj = None
+    if room_id.isdigit():
+        try:
+            current_room_obj = Room.objects.get(id=int(room_id))
+        except Room.DoesNotExist:
+            pass
+
+    # Fetch all rooms checked in for the given round, ordered by rank
+    # A little out of depth on the Django ORM here, but this seems like the
+    # way to get this data with the least amount of queries
+    rooms = set(Room.objects.filter(
+        roomcheckin__round_number=round_number
+    ).annotate(
+        has_round=Exists(Round.objects.filter(room_id=OuterRef('id')))
+    ).order_by("-rank"))
+
+    
+    
+    """
+    There are three relavent pieces of information tab staff may need about rooms
+    1. Is it checked in?
+    2. Is it already paired?
+    3. Is it viable (meets all the tags of the round)?
+    
+    Right now, #1 is filtered and #2 and #3 are displayed via separate lists
+    The rationale is that allowing you to select an unchecked in room either
+    a. requires you to check it in, which could accidentally check in a room that shouldn't be
+    b. not check it in, which would break the rule of paired rooms needing a room checkin
+    
+    I think displaying unchecked in rooms too might be helpful, but only if its implimented
+    with a pop-up to check in the room, which was overhead I didn't want to include in this
+    slate of features.
+    """
+
+    # For now all checked in rooms are "viable", but room tags will make having three categories nessicary
+    viable_rooms = rooms
+    
+    viable_unpaired_rooms = filter(lambda room: not room.has_round, rooms)
+    viable_paired_rooms = filter(lambda room: room.has_round, rooms)
+    other_rooms = rooms - viable_rooms
+
+    # Render the response
+    return render(request, "pairing/room_dropdown.html", {
+        "current_room": current_room_obj,
+        "round_obj": round_obj,
+        "viable_rooms": viable_rooms,
+        "viable_paired_rooms": viable_paired_rooms,
+        "other_rooms": other_rooms,
+    })
+    
+@permission_required("tab.tab_settings.can_change", login_url="/403/")
+def assign_room(request, round_id, room_id, remove_id=None):
+    try:
+        round_obj = Round.objects.get(id=int(round_id))
+        if remove_id is not None and round_obj.room.id != remove_id:
+                round_obj.room = None
+        room_obj = Room.objects.get(id=int(room_id))
+        round_obj.room = room_obj
+        round_obj.save()
+        room_color = room_obj.get_color()
+        data = {
+            "success": True,
+            "room_color": room_color if room_color else "",
+            "room_id": room_obj.id,
+            "round_id": round_obj.id,
+            "room_name": room_obj.name,
+        }
+    except Exception:
+        emit_current_exception()
+        data = {"success": False}
+    return JsonResponse(data)
```

Comment:
nit: if we can make this response have an error code (as opposed to returning a 2XX) it'll be slightly more idiomatic